// ng2-charts/esm2015/lib/ng-charts.module.js
import { Injectable, NgModule, Optional, SkipSelf } from '@angular/core';
import { BaseChartDirective } from './base-chart.directive';
import { ArcElement, BarController, BarElement, BubbleController, CategoryScale, Chart, defaults, DoughnutController, Filler, Legend, LinearScale, LineController, LineElement, PieController, PointElement, PolarAreaController, RadarController, RadialLinearScale, ScatterController, TimeSeriesScale, Title, Tooltip } from 'chart.js';
import { merge } from "lodash-es";
import { builtInDefaults } from "./get-colors";
import * as i0 from "@angular/core";
Chart.register(Title, Tooltip, Filler, Legend, LineController, LineElement, PointElement, LinearScale, CategoryScale, BarController, BarElement, DoughnutController, ArcElement, RadarController, RadialLinearScale, PieController, PolarAreaController, BubbleController, ScatterController, TimeSeriesScale);
export class NgChartsModule {
    constructor(config, parentModule) {
        if (parentModule) {
            throw new Error('NgChartsModule is already loaded. Use .forRoot() in the AppModule only');
        }
        if (config === null || config === void 0 ? void 0 : config.plugins)
            Chart.register(config === null || config === void 0 ? void 0 : config.plugins);
        const ngChartsDefaults = merge(builtInDefaults, (config === null || config === void 0 ? void 0 : config.defaults) || {});
        defaults.set(ngChartsDefaults);
    }
    static forRoot(config) {
        return {
            ngModule: NgChartsModule,
            providers: [
                { provide: NgChartsConfiguration, useValue: config }
            ]
        };
    }
}
NgChartsModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [BaseChartDirective],
                exports: [BaseChartDirective]
            },] }
];
NgChartsModule.ctorParameters = () => [
    { type: NgChartsConfiguration, decorators: [{ type: Optional }] },
    { type: NgChartsModule, decorators: [{ type: Optional }, { type: SkipSelf }] }
];
export class NgChartsConfiguration {
}
NgChartsConfiguration.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgChartsConfiguration_Factory() { return new NgChartsConfiguration(); }, token: NgChartsConfiguration, providedIn: NgChartsModule });
NgChartsConfiguration.decorators = [
    { type: Injectable, args: [{ providedIn: NgChartsModule },] }
];
