// ng2-charts/schematics/ng2-charts-schematics/ng2-charts-line.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ng2ChartsLine = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const ng2_process_tree_1 = require("./ng2-process-tree");
const build_meta_config_1 = require("./build-meta-config");
const newCode = `public lineChartData: ChartDataSetsLine[] = [
    { data: [65, 59, 80, 81, 56, 55, 40], label: 'Series A' },
  ];
  public lineChartLabels: Label[] = ['January', 'February', 'March', 'April', 'May', 'June', 'July'];
  public lineChartOptions: ChartOptions = {
    responsive: true,
  };
  public lineChartColors: Color[] = [
    {
      borderColor: 'black',
      backgroundColor: 'rgba(255,0,0,0.3)',
    },
  ];
  public lineChartLegend = true;
  public lineChartType: ChartType = 'line';
  public lineChartPlugins = [];

  constructor() { }

  ngOnInit(): void {
  }`;
const newMarkup = `<div style="display: block;">
  <canvas baseChart width="400" height="400"
    [data]="lineChartData"
    [labels]="lineChartLabels"
    [options]="lineChartOptions"
    [colors]="lineChartColors"
    [legend]="lineChartLegend"
    [type]="lineChartType"
    [plugins]="lineChartPlugins">
  </canvas>
</div>
`;
const newImports = [
    ['ChartDataSetsLine, ChartOptions, ChartType, Color, Label', 'src/app/app-chart-config'],
];
// You don't have to export the function as default. You can also have more than one rule factory
// per file.
function ng2ChartsLine(options) {
    return schematics_1.chain([
        schematics_1.externalSchematic('@schematics/angular', 'component', options),
        build_meta_config_1.buildMetaConfig(options),
        ng2_process_tree_1.ng2ProcessTree(options, newCode, newMarkup, newImports),
    ]);
}
exports.ng2ChartsLine = ng2ChartsLine;
