// ng2-charts/schematics/ng2-charts-schematics/ng2-charts-pie.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ng2ChartsPie = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const ng2_process_tree_1 = require("./ng2-process-tree");
const build_meta_config_1 = require("./build-meta-config");
const newCode = `public pieChartOptions: ChartOptions = {
    responsive: true,
  };
  public pieChartLabels: Label[] = ['Download Sales', 'In-Store Sales', 'Mail Sales'];
  public pieChartData: SingleDataSet = [300, 500, 100];
  public pieChartType: ChartType = 'pie';
  public pieChartLegend = true;
  public pieChartPlugins = [];

  constructor() { }

  ngOnInit(): void {
  }`;
const newMarkup = `<div style="display: block;">
  <canvas baseChart
    [data]="pieChartData"
    [labels]="pieChartLabels"
    [type]="pieChartType"
    [options]="pieChartOptions"
    [plugins]="pieChartPlugins"
    [legend]="pieChartLegend">
  </canvas>
</div>
`;
const newImports = [
    ['ChartOptions, ChartType, Label, SingleDataSet', 'src/app/app-chart-config'],
];
// You don't have to export the function as default. You can also have more than one rule factory
// per file.
function ng2ChartsPie(options) {
    return schematics_1.chain([
        schematics_1.externalSchematic('@schematics/angular', 'component', options),
        build_meta_config_1.buildMetaConfig(options),
        ng2_process_tree_1.ng2ProcessTree(options, newCode, newMarkup, newImports)
    ]);
}
exports.ng2ChartsPie = ng2ChartsPie;
