// ng2-charts/schematics/ng2-charts-schematics/ng2-charts-bubble.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ng2ChartsBubble = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const ng2_process_tree_1 = require("./ng2-process-tree");
const build_meta_config_1 = require("./build-meta-config");
const newCode = `public bubbleChartOptions: ChartOptions = {
    responsive: true,
    scales: {
      xAxes: [{
        ticks: {
          min: 0,
          max: 30,
        }
      }],
      yAxes: [{
        ticks: {
          min: 0,
          max: 30,
        }
      }]
    }
  };
  public bubbleChartType: ChartType = 'bubble';
  public bubbleChartLegend = true;

  public bubbleChartData: ChartDataSetsBubble[] = [
    {
      data: [
        { x: 10, y: 10, r: 10 },
        { x: 15, y: 5, r: 15 },
        { x: 26, y: 12, r: 23 },
        { x: 7, y: 8, r: 8 },
      ],
      label: 'Series A',
    },
  ];

  constructor() { }

  ngOnInit(): void {
  }`;
const newMarkup = `<div style="display: block;">
  <canvas baseChart
    [data]="bubbleChartData"
    [options]="bubbleChartOptions"
    [colors]="bubbleChartColors"
    [legend]="bubbleChartLegend"
    [type]="bubbleChartType">
  </canvas>
</div>
`;
const newImports = [
    ['ChartDataSetsBubble, ChartOptions, ChartType, Color', 'src/app/app-chart-config'],
];
// You don't have to export the function as default. You can also have more than one rule factory
// per file.
function ng2ChartsBubble(options) {
    return schematics_1.chain([
        schematics_1.externalSchematic('@schematics/angular', 'component', options),
        build_meta_config_1.buildMetaConfig(options),
        ng2_process_tree_1.ng2ProcessTree(options, newCode, newMarkup, newImports)
    ]);
}
exports.ng2ChartsBubble = ng2ChartsBubble;
