// ng2-charts/schematics/ng2-charts-schematics/ng2-charts-scatter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ng2ChartsScatter = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const ng2_process_tree_1 = require("./ng2-process-tree");
const build_meta_config_1 = require("./build-meta-config");
const newCode = `public scatterChartOptions: ChartOptions = {
    responsive: true,
  };

  public scatterChartData: ChartDataSetsScatter[] = [
    {
      data: [
        { x: 1, y: 1 },
        { x: 2, y: 3 },
        { x: 3, y: -2 },
        { x: 4, y: 4 },
        { x: 5, y: -3, r: 20 },
      ],
      label: 'Series A',
      pointRadius: 10,
    },
  ];
  public scatterChartType: ChartType = 'scatter';

  constructor() { }

  ngOnInit(): void {
  }`;
const newMarkup = `<div style="display: block;">
  <canvas baseChart
    [data]="scatterChartData"
    [options]="scatterChartOptions"
    [type]="scatterChartType">
  </canvas>
</div>`;
const newImports = [
    ['ChartDataSetsScatter, ChartOptions, ChartType, Label', 'src/app/app-chart-config'],
];
// You don't have to export the function as default. You can also have more than one rule factory
// per file.
function ng2ChartsScatter(options) {
    return schematics_1.chain([
        schematics_1.externalSchematic('@schematics/angular', 'component', options),
        build_meta_config_1.buildMetaConfig(options),
        ng2_process_tree_1.ng2ProcessTree(options, newCode, newMarkup, newImports)
    ]);
}
exports.ng2ChartsScatter = ng2ChartsScatter;
