// ng2-charts/schematics/ng2-charts-schematics/ng2-charts-radar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ng2ChartsRadar = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const ng2_process_tree_1 = require("./ng2-process-tree");
const build_meta_config_1 = require("./build-meta-config");
const newCode = `public radarChartOptions: ChartOptions = {
    responsive: true,
  };
  public radarChartLabels: Label[] = ['Eating', 'Drinking', 'Sleeping', 'Designing', 'Coding', 'Cycling', 'Running'];

  public radarChartData: ChartDataSetsRadar[] = [
    { data: [65, 59, 90, 81, 56, 55, 40], label: 'Series A' },
    { data: [28, 48, 40, 19, 96, 27, 100], label: 'Series B' }
  ];
  public radarChartType: ChartType = 'radar';

  constructor() { }

  ngOnInit(): void {
  }`;
const newMarkup = `<div style="display: block;">
  <canvas baseChart
    [data]="radarChartData"
    [options]="radarChartOptions"
    [labels]="radarChartLabels"
    [type]="radarChartType">
  </canvas>
</div>
`;
const newImports = [
    ['ChartDataSetsRadar, ChartOptions, ChartType, Label', 'src/app/app-chart-config'],
];
// You don't have to export the function as default. You can also have more than one rule factory
// per file.
function ng2ChartsRadar(options) {
    return schematics_1.chain([
        schematics_1.externalSchematic('@schematics/angular', 'component', options),
        build_meta_config_1.buildMetaConfig(options),
        ng2_process_tree_1.ng2ProcessTree(options, newCode, newMarkup, newImports)
    ]);
}
exports.ng2ChartsRadar = ng2ChartsRadar;
